index.html

```diff
@@ -259,10 +259,200 @@ <h1>Architecture</h1>
+	its Introduction phase, even if only one Introduction mechanism is used,
+	and the final output after the Exploration phase
+	can be a <em>set</em> of Thing Descriptions.
+	Generally each URL provided by the Introduction phase points at a Thing Description,
+	and in the simplest case this is in fact the Thing Description of a Thing that
+	represents an IoT device.  In the case of self-describing Things the URL might
+	even point directly at the Thing which can serve its own Thing Description,
+	but in general Thing Descriptions might also be provided by other entities.
+	However, there are also two special cases that allow for more flexibility:
+	</p>
+	<ul>
+	<li>A given Thing Description may in fact describe a Thing Description Directory service.  
+	A Thing Description Directory service (which is still a Thing) maintains a (usually dynamic) list of Thing Descriptions
+	which can be fetched and added to the discovery set (the set can be narrowed by
+	using query filters when fetching TDs from a directory).
+	</li>
+	<li>
+	The second special case is a Thing Link.  This is also a Thing Description,
+	but rather than describing a Thing directly holds a link to
+	a Thing Descriptions hosted elsewhere.  
+	A Thing Description Directory can also store Thing Links
+	which can redirect to other Thing Description Directories, allowing for a linked
+	directory structure.
+	</li>
+	</ul>
+    </section>
+    <section id="consumer-process" class="normative">
+        <h1>Clients and Consumers</h1>
+	<p>In this section we will describe the WoT Discovery process from the point of view of a client,
+	and what it means to say that a client supports WoT Discovery.
+	We will define a Discovering Client, D-Client for short, as an entity that supports the WoT Discovery
+	process.  Such a client may or may not be a Consumer.  A D-Client finds TDs, which are then passed
+	onto a Consumer for interpretation.  A D-Client itself may do some interpretation of TDs that it
+	fetches, however, in particular to examine the types of the TD it fetches to see if additional
+	fetches are required.  A Consumer may also be D-Client, in which case we will call
+	it a Discovering Consumer, or D-Consumer.
+	All of the statements below about D-Clients also apply to D-Consumers.
+
+	These requirements are designed so that nearly any client that can fetch a single TD 
+	given a single URL over any protocol (including "file") can be said to support WoT Discovery. 
+	Of course, clients may support more powerful Discovery mechanisms, and some of these 
+	have their own requirements.  
+	For example, using WoT Thing Description Directories (TDDs) requires the use of HTTP, and so
+	are only available to clients that support this protocol.  However, use of HTTP (and Directories)
+	is not in fact a requirement of the WoT Discovery process: it is possible to discover multiple
+	TDs using just Introduction mechanisms.
```

Comment:
Getting the TD is part of exploration. That is possible with or without a directory.

Reply:
Ah, true.  Wording needs some tweaking.  Should say "without using Directories".

Reply:
reworded in update